ledger/accountdb.go

```diff
@@ -4151,57 +4201,62 @@ func processAllResources(
 			// in this case addrid = 3 after processing resources from 1, but acctRowid = 2
 			// and we need to skip accounts without resources
 			if pr.addrid > acctRowid {
-				err = callback(addr, 0, nil, nil)
-				return pr, err
+				err = callback(addr, 0, nil, nil, false)
+				return pr, count, err
 			}
 			if pr.addrid < acctRowid {
 				err = fmt.Errorf("resource table entries mismatches accountbase table entries : reached addrid %d while expecting resource for %d", pr.addrid, acctRowid)
-				return pendingRow{}, err
+				return pendingResourceRow{}, count, err
 			}
 			addrid = pr.addrid
 			buf = pr.buf
 			aidx = pr.aidx
-			pr = pendingRow{}
+			pr = pendingResourceRow{}
 		} else {
 			if !resRows.Next() {
-				err = callback(addr, 0, nil, nil)
+				err = callback(addr, 0, nil, nil, false)
 				if err != nil {
-					return pendingRow{}, err
+					return pendingResourceRow{}, count, err
 				}
 				break
 			}
 			err = resRows.Scan(&addrid, &aidx, &buf)
 			if err != nil {
-				return pendingRow{}, err
+				return pendingResourceRow{}, count, err
 			}
 			if addrid < acctRowid {
 				err = fmt.Errorf("resource table entries mismatches accountbase table entries : reached addrid %d while expecting resource for %d", addrid, acctRowid)
-				return pendingRow{}, err
+				return pendingResourceRow{}, count, err
 			} else if addrid > acctRowid {
-				err = callback(addr, 0, nil, nil)
-				return pendingRow{addrid, aidx, buf}, err
+				err = callback(addr, 0, nil, nil, false)
+				return pendingResourceRow{addrid, aidx, buf}, count, err
 			}
 		}
 		resData = resourcesData{}
 		err = protocol.Decode(buf, &resData)
 		if err != nil {
-			return pendingRow{}, err
+			return pendingResourceRow{}, count, err
 		}
-		err = callback(addr, aidx, &resData, buf)
+		count++
+		if resourceCount > 0 && count == resourceCount {
```

Comment:
maybe a comment here to explain what this callback(..., true) is for

Reply:
+1 need a comment